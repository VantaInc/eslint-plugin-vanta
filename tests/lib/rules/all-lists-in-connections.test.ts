import { GraphQLRuleTester } from "@graphql-eslint/eslint-plugin";
import rule from "../../../lib/rules/all-lists-in-connections";

const ruleTester = new GraphQLRuleTester();

ruleTester.runGraphQLTests("all-lists-in-connections", rule, {
  valid: [
    {
      code: `
type T {
  users: Int!
}
`,
    },
    {
      code: `
type T {
  users: UserConnection
}

type UserConnection {
  pageInfo: PageInfo!
  edges: [UserEdge!]!
  totalCount: Int!
}

type PageInfo {
  endCursor: String
  hasNextPage: Boolean!
  hasPreviousPage: Boolean!
  startCursor: String
}

type UserEdge {
  cursor: String!
  node: User!
}

type User {
  id: ID!
  name: String
}
`,
    },
    {
      code: `
type T {
  users: [User!]! @tinylist
}

type User {
  id: ID!
  name: String
}
`,
    },
  ],
  invalid: [
    {
      code: `
type T {
  users: [User!]!
}

type User {
  id: ID!
  name: String
}
`,
      errors: 1,
    },
    {
      code: `
type T {
  users: [User!]!
  users1: [User]
}

type User {
  id: ID!
  name: String
}
`,
      errors: 2,
    },
    {
      code: `
type T {
  users: [User!]! @tinylist
  users1: [User!]
}

type User {
  id: ID!
  name: String
}
`,
      errors: 1,
    },
    {
      code: `
type T {
  users: UserConnection
}

type UserConnection {
  pageInfo: PageInfo!
  edges: [UserEdge!]!
  notEdges: [UserEdge!]! # this shouldn't be a list
  totalCount: Int!
}

type PageInfo {
  endCursor: String
  hasNextPage: Boolean!
  hasPreviousPage: Boolean!
  startCursor: String
}

type UserEdge {
  cursor: String!
  node: User!
}

type User {
  id: ID!
  name: String
}
`,
      errors: 1,
    },
  ],
});
