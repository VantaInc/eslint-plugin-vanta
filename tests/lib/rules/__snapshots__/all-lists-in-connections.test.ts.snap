// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid #1 1`] = `
"#### ⌨️ Code

      1 | type T {
      2 |   users: [User!]!
      3 | }
      4 |
      5 | type User {
      6 |   id: ID!
      7 |   name: String
      8 | }

#### ❌ Error

      1 | type T {
    > 2 |   users: [User!]!
        |   ^^^^^^^^^^^^^^ Lists must be paginated or marked as guaranteed to be short using the directive @tinylist.
      3 | }"
`;

exports[`Invalid #2 1`] = `
"#### ⌨️ Code

      1 | type T {
      2 |   users: [User!]!
      3 |   users1: [User]
      4 | }
      5 |
      6 | type User {
      7 |   id: ID!
      8 |   name: String
      9 | }

#### ❌ Error 1/2

      1 | type T {
    > 2 |   users: [User!]!
        |   ^^^^^^^^^^^^^^ Lists must be paginated or marked as guaranteed to be short using the directive @tinylist.
      3 |   users1: [User]

#### ❌ Error 2/2

      2 |   users: [User!]!
    > 3 |   users1: [User]
        |   ^^^^^^^^^^^^^ Lists must be paginated or marked as guaranteed to be short using the directive @tinylist.
      4 | }"
`;

exports[`Invalid #3 1`] = `
"#### ⌨️ Code

      1 | type T {
      2 |   users: [User!]! @tinylist
      3 |   users1: [User!]
      4 | }
      5 |
      6 | type User {
      7 |   id: ID!
      8 |   name: String
      9 | }

#### ❌ Error

      2 |   users: [User!]! @tinylist
    > 3 |   users1: [User!]
        |   ^^^^^^^^^^^^^^ Lists must be paginated or marked as guaranteed to be short using the directive @tinylist.
      4 | }"
`;

exports[`Invalid #4 1`] = `
"#### ⌨️ Code

       1 | type T {
       2 |   users: UserConnection
       3 | }
       4 |
       5 | type UserConnection {
       6 |   pageInfo: PageInfo!
       7 |   edges: [UserEdge!]!
       8 |   notEdges: [UserEdge!]! # this shouldn't be a list
       9 |   totalCount: Int!
      10 | }
      11 |
      12 | type PageInfo {
      13 |   endCursor: String
      14 |   hasNextPage: Boolean!
      15 |   hasPreviousPage: Boolean!
      16 |   startCursor: String
      17 | }
      18 |
      19 | type UserEdge {
      20 |   cursor: String!
      21 |   node: User!
      22 | }
      23 |
      24 | type User {
      25 |   id: ID!
      26 |   name: String
      27 | }

#### ❌ Error

      7 |   edges: [UserEdge!]!
    > 8 |   notEdges: [UserEdge!]! # this shouldn't be a list
        |   ^^^^^^^^^^^^^^^^^^^^^ Lists must be paginated or marked as guaranteed to be short using the directive @tinylist.
      9 |   totalCount: Int!"
`;
